src/main/java/com/internet/mixshop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,175 @@
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Order order = getOrder(resultSet, connection);
+                return Optional.ofNullable(order);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get order with id = " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Order update(Order order) {
+        String query = "UPDATE orders SET user_id = ?, "
+                + "WHERE order_id = ? AND is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, order.getUserId());
+            statement.setLong(2, order.getId());
+            statement.executeUpdate();
+            return order;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Order " + order.getId() + " can't be updated", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long orderId) {
+        String query = "UPDATE orders SET is_deleted = true WHERE order_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, orderId);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete order with id " + orderId, e);
+        }
+    }
+
+    @Override
+    public List<Order> getAll() {
+        List<Order> orders = new ArrayList<>();
+        String query = "SELECT * FROM orders WHERE is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                orders.add(buildOrder(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can' get orders", e);
+        }
+        for (Order order : orders) {
+            order.setProducts(getProductsListByOrderId(order.getId()));
+        }
+        return orders;
+    }
+
+    private Order getOrder(ResultSet resultSet, Connection connection) throws SQLException {
+        Long orderId = resultSet.getLong("order_id");
+        Long userId = resultSet.getLong("user_id");
+        List<Product> products = getProductsListByOrderId(orderId);
+        return new Order(orderId, products, userId);
+    }
+
+    private void addProductsToOrder(List<Product> products, Long orderId) throws SQLException {
+        String query = "INSERT INTO orders_products (order_id, product_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
```

Comment:
Ideally, you should also close a previous statement before opening a new statement, but that may require changing a lot of code, and you could try to do it after you deal with the connections)